gdal/port/cpl_odbc.cpp

```diff
@@ -845,7 +865,7 @@ int CPLODBCStatement::Fetch( int nOrientation, int nOffset )
 
         if( Failed( nRetCode ) )
         {
-            if ( nRetCode == SQL_NO_DATA )
+            if ( nRetCode != SQL_NO_DATA )
```

Comment:
This line was modified, because network errors during `Fetch()` weren't reported. It seams to me, that the handling of the return code ( `nRetCode == SQL_NO_DATA`) might be problematic in other places as well, however, I couldn't test them.